compilers/concrete-optimizer/concrete-optimizer/src/optimization/dag/multi_parameters/symbolic_variance.rs

```diff
@@ -37,12 +39,12 @@ impl SymbolicVariance {
 
     pub fn nan(nb_partitions: usize) -> Self {
         Self {
-            partition: usize::MAX,
+            partition: PartitionIndex::INVALID,
             coeffs: OperationsValue::nan(nb_partitions),
         }
     }
 
-    pub fn input(nb_partitions: usize, partition: usize) -> Self {
+    pub fn input(nb_partitions: usize, partition: PartitionIndex) -> Self {
```

Comment:
maybe using impl Into<PartiionIndex> would work with both integer and partitionIndex to avoid all the extra range conversions.

Reply:
I prefer not to. Forcing the user to put a `PartitionIndex` annotation everywhere makes it simpler to know at a given location what is what. There is a lot of usize indexes everywhere in the optimizer and IMHO it makes it very difficult to track what is what.